test/wallet-rpc-test.js

```diff
@@ -45,17 +48,24 @@ const wclient = new WalletClient({
   apiKey: 'bar'
 });
 
+const defaultCoinbaseMaturity = consensus.COINBASE_MATURITY;
+
 describe('Wallet RPC Methods', function() {
   this.timeout(15000);
 
   // Define an account level hd extended public key to be
   // used to derive addresses throughout the test suite
   let xpub;
 
+  let walletHot = null;
+  let walletMiner = null;
+  let addressHot = null;
+  let addressMiner = null;
+  let utxo = null;
+
   before(async () => {
+    consensus.COINBASE_MATURITY = 0;
```

Comment:
There is a chance this isn't restored if any of the below fail within this test, it'll probably be preferable to quickly generate more than 100 blocks, at the start of the tests and avoid modifying globals.

Reply:
May be another follow-up that is necessary, there may be other such cases.

Reply:
yeah, this can happen if `before` fails and can affect other test cases.

Ideally consensus should be accepted as parameters in places that it's necessary instead of being used as globals. That would make testing much easier and more flexible.

Generating 100 blocks is not too much to ask but slows downs generation quite a lot, in case we have many separate test cases it may become burden. We could have "mock" chain that already has 100 blocks in one way or another backed up in `test/data`

Reply:
Generating 200 blocks takes around 400-180ms, as from some other tests.

Reply:
If it's done a lot, it could add up, but generally there can be setup scripts that build out a chain, and many tests that run against it.